api/backend.go

```diff
@@ -238,6 +242,32 @@ func (b *StatusBackend) CallRPC(inputJSON string) (string, error) {
 	return client.CallRaw(inputJSON), nil
 }
 
+// GetNodesFromContract returns a list of mailservers
+func (b *StatusBackend) GetNodesFromContract(rpcEndpoint string, contractAddress string) ([]string, error) {
+	var response []string
+
+	ctx, cancel := context.WithTimeout(context.Background(), contractQueryTimeout)
+	defer cancel()
+
+	ethclient, err := ethclient.DialContext(ctx, rpcEndpoint)
+	if err != nil {
+		return response, err
+	}
+
+	contract, err := registry.NewNodes(gethcommon.HexToAddress(contractAddress), ethclient)
+	if err != nil {
+		return response, err
+	}
+
+	node, err := contract.Nodes(nil, big.NewInt(0))
```

Comment:
it wasn't not obvious that you are fetching only the first one, isn't it better to request a certain number using rpc?

Reply:
oh, yes, I am just fetching one for testing as of now, I meant to fetch all but forgot, will add in the next PR!